x-pack/plugin/index-lifecycle/src/main/java/org/elasticsearch/xpack/indexlifecycle/action/TransportMoveToStepAction.java

```diff
@@ -0,0 +1,94 @@
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.core.indexlifecycle.LifecycleSettings;
+import org.elasticsearch.xpack.core.indexlifecycle.action.MoveToStepAction;
+import org.elasticsearch.xpack.core.indexlifecycle.action.MoveToStepAction.Request;
+import org.elasticsearch.xpack.core.indexlifecycle.action.MoveToStepAction.Response;
+import org.elasticsearch.xpack.indexlifecycle.IndexLifecycleRunner;
+import org.elasticsearch.xpack.indexlifecycle.IndexLifecycleService;
+import org.elasticsearch.xpack.indexlifecycle.MoveToNextStepUpdateTask;
+
+public class TransportMoveToStepAction extends TransportMasterNodeAction<Request, Response> {
+    IndexLifecycleService indexLifecycleService;
+    @Inject
+    public TransportMoveToStepAction(Settings settings, TransportService transportService, ClusterService clusterService,
+                                     ThreadPool threadPool, ActionFilters actionFilters,
+                                     IndexNameExpressionResolver indexNameExpressionResolver,
+                                     IndexLifecycleService indexLifecycleService) {
+        super(settings, MoveToStepAction.NAME, transportService, clusterService, threadPool, actionFilters, indexNameExpressionResolver,
+                Request::new);
+        this.indexLifecycleService = indexLifecycleService;
+    }
+
+    @Override
+    protected String executor() {
+        return ThreadPool.Names.SAME;
+    }
+
+    @Override
+    protected Response newResponse() {
+        return new Response();
+    }
+
+    @Override
+    protected void masterOperation(Request request, ClusterState state, ActionListener<Response> listener) {
+        IndexMetaData indexMetaData = state.metaData().index(request.getIndex());
+        if (indexMetaData == null) {
+            listener.onFailure(new IllegalArgumentException("index [" + request.getIndex() + "] does not exist"));
+            return;
+        }
+        String policy = LifecycleSettings.LIFECYCLE_NAME_SETTING.get(indexMetaData.getSettings());
+        if (Strings.isNullOrEmpty(policy)) {
+            listener.onFailure(new IllegalArgumentException("index [" + request.getIndex()
+                + "] is not associated with a lifecycle policy"));
+            return;
+        }
+
+        MoveToNextStepUpdateTask innerTask = new MoveToNextStepUpdateTask(indexMetaData.getIndex(), policy, request.getCurrentStepKey(),
```

Comment:
Can we put this logic in a method on `IndexLifecycleRunner` and then expose it here via `IndexLifecycleService`? I would like to keep all the logic around moving steps in `IndexLifecycleRunner` but I think the action should interact with the `IndexLifecycleService`

Reply:
for sure, I'll move this to bubble up from the IndexLifecycleService. I wanted to re-use the task is all.